code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -100,14 +100,16 @@
 	add_fingerprint(user)
 
 /obj/item/weapon/melee/baton/attack(mob/M, mob/user)
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN*1.5)
 	if(status && (CLUMSY in user.mutations) && prob(50))
 		user << "<span class='danger'>You accidentally hit yourself with the [src]!</span>"
 		user.Weaken(30)
 		deductcharge(hitcost)
 		return
 	return ..()
 
-/obj/item/weapon/melee/baton/attack(mob/M, mob/user,var/hit_zone)
+/obj/item/weapon/melee/baton/attack(mob/M, mob/user, var/hit_zone)
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN*1.5)
```

Comment:
This should be `user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)`. Whoever thought that we should suddenly start applying different cooldowns based on weapons is silly. Nothing else does it, and without actually planning to apply it to other weaponry, we aren't starting.

Also, since setClickCooldown was missing, it was probably also missing this: `user.do_attack_animation(target)`.

Second also. Two attack proc definitions = fucked up call stack. Remove the top one as it's not used, and will most likely result in even more cooldown being applied.

Third also: move the call below the `isrobot(M)` clause. Requires testing, but if I'm correct, the `..()` call there will apply the cooldown. It should actually go into the `else` clause for `if(user.a_intent == I_HURT)`, as that if will **also** apply the cooldown.